solidity/test/bridge/Bridge.Frauds.test.ts

```diff
@@ -0,0 +1,1528 @@
+              ecdsaWalletID: ethers.constants.HashZero,
+              mainUtxoHash: ethers.constants.HashZero,
+              pendingRedemptionsValue: 0,
+              createdAt: await lastBlockTime(),
+              state: walletState.Live,
+            })
+          })
+
+          after(async () => {
+            await restoreSnapshot()
+          })
+
+          it("should revert", async () => {
+            await expect(
+              bridge.connect(thirdParty).submitFraudChallenge(
+                walletPublicKey,
+                data.sighash,
+                {
+                  r: incorrectR,
+                  s: incorrectS,
+                  v: data.signature.v,
+                },
+                {
+                  value: await bridge.fraudChallengeDepositAmount(),
+                }
+              )
+            ).to.be.revertedWith("Signature verification failure")
+          })
+        })
+      })
+
+      context("when the amount of ETH deposited is too low", () => {
+        before(async () => {
+          await createSnapshot()
+          await bridge.setWallet(walletPublicKeyHash, {
+            ecdsaWalletID: ethers.constants.HashZero,
+            mainUtxoHash: ethers.constants.HashZero,
+            pendingRedemptionsValue: 0,
+            createdAt: await lastBlockTime(),
+            state: walletState.Live,
+          })
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should revert", async () => {
+          await expect(
+            bridge
+              .connect(thirdParty)
+              .submitFraudChallenge(
+                walletPublicKey,
+                data.sighash,
+                data.signature,
+                {
+                  value: (await bridge.fraudChallengeDepositAmount()).sub(1),
+                }
+              )
+          ).to.be.revertedWith("The amount of ETH deposited is too low")
+        })
+      })
+    })
+
+    context("when the wallet is in MovingFunds state", () => {
+      before(async () => {
+        await createSnapshot()
+        await bridge.setWallet(walletPublicKeyHash, {
+          ecdsaWalletID: ethers.constants.HashZero,
+          mainUtxoHash: ethers.constants.HashZero,
+          pendingRedemptionsValue: 0,
+          createdAt: await lastBlockTime(),
+          state: walletState.Live,
+        })
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+
+      it("should succeed", async () => {
+        await expect(
+          bridge
+            .connect(thirdParty)
+            .submitFraudChallenge(
+              walletPublicKey,
+              data.sighash,
+              data.signature,
+              {
+                value: await bridge.fraudChallengeDepositAmount(),
+              }
+            )
+        ).to.not.be.reverted
+      })
+    })
+
+    context("when the wallet is in neither Live nor MovingFunds state", () => {
```

Comment:
Worth testing all states other than `Live` and `MovingFunds`. Here is an useful pattern: https://github.com/keep-network/tbtc-v2/blob/e5d80c27c0282ae9f153619148bc86c5364e5424/solidity/test/bridge/Bridge.Wallets.test.ts#L1399

Reply:
Done in 2321ab635cc0cf8890dd0edf6f11550774a8300c.